### ISSUE TITLE
Peripheral half crashes when USB Logging is enabled

### ISSUE THREAD

**Issue by kurtis-lew** [zmkfirmware/zmk#1245]
Suspected to be related to [#44349 on the Zephyr Project](https://github.com/zephyrproject-rtos/zephyr/issues/44349).

Peripheral half crashes after consecutive keypresses occur too quickly. Discussion of this problem on Discord begins [here](https://discord.com/channels/719497620560543766/719544230497878116/963448190370189312)

The working theory is that there is an issue with the [`bt_gatt_notify` call in `app/src/split/bluetooth/service.c`](https://github.com/zmkfirmware/zmk/blob/main/app/src/split/bluetooth/service.c#L114-L117). Rather than requiring sufficiently fast user input, attempted fixes to this issue, like commenting out or changing `LOG_DBG` to `LOG_WRN`/`LOG_ERR` in the lines linked previously, changing the logging level in the [`LOG_MODULE_DECLARE` statement](https://github.com/zmkfirmware/zmk/blob/c7a6836735d0a77d8327d4377ebacd0e25c5bb9e/app/src/split/bluetooth/service.c#L13) to 0, or adding `CONFIG_BT_BUF_ACL_TX_COUNT=32` to the peripheral's config file, causes the peripheral to crash immediately after bootup logs.

**Comment by kurtis-lew** [comment]
A method suggested for debugging by @xudongzheng was to use [`LOG_HEXDUMP_DBG`](https://discord.com/channels/719497620560543766/719544230497878116/963906410100449350) instead of lines 114-117 in `app/src/split/bluetooth/service.c`. When using the hexdump logging, we see that the peripheral half can output logs, but its keypresses are not invoked properly.
```
[00:00:30.275,939] [0m<dbg> zmk: zmk_usb_get_conn_state: state: 3[0m
[00:00:33.286,132] [0m<dbg> zmk: kscan_matrix_read: Sending event at 2,1 state on[0m
[00:00:33.286,224] [0m<dbg> zmk: zmk_kscan_process_msgq: Row: 2, col: 1, position: 31, pressed: true[0m
[00:00:33.286,254] [0m<dbg> zmk: split_listener: [0m
[00:00:33.286,315] [0m<dbg> zmk: send_position_state_callback: dump
                              00 00 00 80 00 00 00 00  00 00 00 00 00 00 00 00 |........ ........[0m
[00:00:33.352,142] [0m<dbg> zmk: kscan_matrix_read: Sending event at 2,1 state off[0m
[00:00:33.352,264] [0m<dbg> zmk: zmk_kscan_process_msgq: Row: 2, col: 1, position: 31, pressed: false[0m
[00:00:33.352,294] [0m<dbg> zmk: split_listener: [0m
[00:00:33.352,355] [0m<dbg> zmk: send_position_state_callback: dump
                              00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 |........ ........[0m
[00:00:41.475,128] [0m<dbg> zmk: kscan_matrix_read: Sending event at 2,1 state on[0m
[00:00:41.475,219] [0m<dbg> zmk: zmk_kscan_process_msgq: Row: 2, col: 1, position: 31, pressed: true[0m
[00:00:41.475,250] [0m<dbg> zmk: split_listener: [0m
[00:00:41.475,311] [0m<dbg> zmk: send_position_state_callback: dump
                              00 00 00 80 00 00 00 00  00 00 00 00 00 00 00 00 |........ ........[0m
[00:00:41.540,130] [0m<dbg> zmk: kscan_matrix_read: Sending event at 2,1 state off[0m
[00:00:41.540,252] [0m<dbg> zmk: zmk_kscan_process_msgq: Row: 2, col: 1, position: 31, pressed: false[0m
[00:00:41.540,283] [0m<dbg> zmk: split_listener: [0m
[00:00:41.540,344] [0m<dbg> zmk: send_position_state_callback: dump
                              00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 |........ ........[0m
```

**Comment by kurtis-lew** [comment]
Addressed in #1261
